src/test/groovy/org/prebid/server/functional/tests/module/pbrequestcorrection/PbRequestCorrectionSpec.groovy

```diff
@@ -0,0 +1,437 @@
+    }
+
+    def "PBS shouldn't remove pattern device.ua when request correction is enabled for account and user agent correction disabled"() {
+        given: "Android APP bid request with version lover then 2.1.5"
+        def deviceUserAgent = "${DEVICE_PREBID_MOBILE_PATTERN}${PBSUtils.randomNumber}"
+        def prebid = new AppPrebid(source: PBSUtils.getRandomCase(PREBID_MOBILE), version: getRandomVersion("0.0.0", "2.1.5"))
+        def bidRequest = BidRequest.getDefaultBidRequest(APP).tap {
+            app.ext = new AppExt(prebid: prebid)
+            device = new Device(ua: deviceUserAgent)
+        }
+
+        and: "Account in the DB"
+        def requestCorrectionConfig = PbRequestCorrectionConfig.getDefaultConfigWithUserAgentCorrection(userAgentCorrectionEnabled, enabled)
+        def account = createAccountWithRequestCorrectionConfig(bidRequest, requestCorrectionConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        pbsServiceWithRequestCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain device.ua"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.device.ua == deviceUserAgent
+
+        where:
+        enabled | userAgentCorrectionEnabled
+        false   | true
+        null    | true
+        true    | false
+        true    | null
+        null    | null
+    }
+
+    def "PBS shouldn't remove pattern device.ua when request correction is enabled for account and source not a prebid-mobile"() {
+        given: "Android APP bid request with version lover then 2.1.5"
+        def randomDeviceUa = "${DEVICE_PREBID_MOBILE_PATTERN}${PBSUtils.randomNumber}"
+        def bidRequest = BidRequest.getDefaultBidRequest(APP).tap {
+            app.ext = new AppExt(prebid: new AppPrebid(source: source, version: getRandomVersion("0.0.0", "2.1.5")))
+            device = new Device(ua: randomDeviceUa)
+        }
+
+        and: "Account in the DB"
+        def requestCorrectionConfig = PbRequestCorrectionConfig.defaultConfigWithUserAgentCorrection
+        def account = createAccountWithRequestCorrectionConfig(bidRequest, requestCorrectionConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        pbsServiceWithRequestCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain device.ua"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.device.ua == randomDeviceUa
+
+        where:
+        source << ["prebid", "mobile", "mobile-prebid", PBSUtils.randomString]
+    }
+
+    def "PBS shouldn't remove pattern device.ua when request correction is enabled for account and version biggest that 2.1.6"() {
+        given: "Android APP bid request with version lover then 2.1.5"
```

Comment:
`account and version biggest that 2.1.6`
and 
`Android APP bid request with version lover then 2.1.5`
same for others

Reply:
and what about `2.1.6`?